src/heatmap/HeatMapImageRenderer.ts

```diff
@@ -130,11 +134,47 @@ export default class HeatMapImageRenderer extends AHeatMapCanvasRenderer impleme
     } else if (colors.length === 2 || colors.length === 3) {
       args.palette = colors.map(ensureHex).join('-');
     }
-    this.image.src = data.heatmapUrl(all(), args);
+
+    // persist to get range and create range object again
+    // TODO: make range property on matrix public
+    const range = parseRange(data.persist().range);
+    const params = prepareParameter(range, args);
+    const url = `/dataset/matrix/${data.desc.id}/data`;
+
+    const encoded = encodeParams(params);
+    if (encoded && (url.length + encoded.length > TOO_LONG_URL)) {
+      // use post instead
+      sendAPI(url, params, 'POST', 'blob').then((image) => {
+        const imageURL = window.URL.createObjectURL(image);
+        this.image.src = imageURL;
+      });
+    } else {
+      this.image.src = data.heatmapUrl(all(), args);
+    }
 
     super.buildSelection(data, $root, scale);
 
     return $root;
+  }
+}
 
+function prepareParameter(range: Range, options: IHeatMapUrlOptions) {
+  const args: any = {
```

Comment:
is there a typing for the args?

Reply:
We added them with PR phovea/phovea_core#150